packages/@atjson/source-html/src/converter.ts

```diff
@@ -59,5 +59,22 @@ HTMLSource.defineConverterTo(OffsetSource, doc => {
     });
   });
 
+  let $pre = doc.where({ type: '-html-pre' }).as('pre');
+  let $code = doc.where({ type: '-html-code' }).as('codeElements');
+
+  $pre.join($code, (pre, code) => pre.start < code.start && code.end < pre.end).update(({ pre, codeElements }) => {
```

Comment:
This matches more `pre` elements than is strictly correct. For example, we should not replace the annotation if you have a `pre` wrapping a `paragraph` that contains an inline `code`.

You can use an outer join to ensure that the items you are updating are `code` elements that are immediate children of a `pre`, or better, to ensure that the `pre` contains no other immediate children (including text nodes).